README.md
# imei

IMEI, ICCID, and other numbers of your SIM chip and your FirefoxOS phone
![SIM Info for FirefoxOS](http://pix.toile-libre.org/upload/original/1446130575.png)


## Installation

Certified app with heavy use of permissions.
The flashable app lives in /out directory

### Bugs

 * More than 1 SIM is not supported yet
 * MSISDN detection is not working for now


Copyright © 2015 leandro@leandro.org




CHANGELOG.md
# Change Log
All notable changes to this project will be documented in this file. This change log follows the conventions of [keepachangelog.com](http://keepachangelog.com/).

## [Unreleased][unreleased]
### Changed
- Add a new arity to `make-widget-async` to provide a different widget shape.

## [0.1.1] - 2015-10-27
### Changed
- Documentation on how to make the widgets.

### Removed
- `make-widget-sync` - we're all async, all the time.

### Fixed
- Fixed widget maker to keep working when daylight savings switches over.

## 0.1.0 - 2015-10-27
### Added
- Files from the new template.
- Widget maker public API - `make-widget-sync`.

[unreleased]: https://github.com/your-name/imei/compare/0.1.1...HEAD
[0.1.1]: https://github.com/your-name/imei/compare/0.1.0...0.1.1


test/imei/core_test.clj
(ns imei.core-test
  (:require [clojure.test :refer :all]
            [imei.core :refer :all]))

(deftest a-test
  (testing "FIXME, I fail."
    (is (= 0 1))))


project.clj
(defproject imei "0.1.0-SNAPSHOT"
  :description "FIXME: write description"
  :url "http://example.com/FIXME"
  :license {:name "Eclipse Public License"
            :url "http://www.eclipse.org/legal/epl-v10.html"}
    :dependencies [
                    [org.clojure/clojure "1.7.0"]
                    [org.clojure/clojurescript "1.7.145"]
                ]
  :plugins [[lein-cljsbuild "1.1.0"]]
  :hooks [leiningen.cljsbuild]
  :cljsbuild
    {:builds
     [{:source-paths ["src/cljs"]
       :compiler
       {
        :optimizations :whitespace
        :output-to "out/js/imei.js"
        ;:externs ["externs/jquery-1.8.js"]
        :pseudo-names true ;human
        :pretty-print true ;human
        :static-fns true   ;not optimized
        :verbose true      ;debug
        :output-dir "out/js"
        :cache-analysis true
        ;:source-map "out/js/imei-debug.js" ;only for debug
        ;:source-map-path "out/js"  ;only for debug
        ;:source-map-timestamp true ;debug
        }}]}
  :main imei.core
  :target-path "target/%s"
  ;; to clean JS files generated during the build
  :clean-targets ^{:protect false} [:target-path "out/js/"]
  :profiles {:uberjar {:aot :all}})


out/js/constants_table.js
cljs.core.cst$sym$form = new cljs.core.Symbol(null,"form","form",16469056,null);
cljs.core.cst$sym$tag = new cljs.core.Symbol(null,"tag","tag",350170304,null);
cljs.core.cst$sym$_AMPERSAND_ = new cljs.core.Symbol(null,"&","&",-2144855648,null);
cljs.core.cst$sym$stepper = new cljs.core.Symbol(null,"stepper","stepper",1159163296,null);
cljs.core.cst$sym$uuid = new cljs.core.Symbol(null,"uuid","uuid",-504564192,null);
cljs.core.cst$sym$end = new cljs.core.Symbol(null,"end","end",1372345569,null);
cljs.core.cst$sym$defrecord_STAR_ = new cljs.core.Symbol(null,"defrecord*","defrecord*",-1936366207,null);
cljs.core.cst$sym$base = new cljs.core.Symbol(null,"base","base",1825810849,null);
cljs.core.cst$sym$obj = new cljs.core.Symbol(null,"obj","obj",-1672671807,null);
cljs.core.cst$sym$fqn = new cljs.core.Symbol(null,"fqn","fqn",-1749334463,null);
cljs.core.cst$sym$first = new cljs.core.Symbol(null,"first","first",996428481,null);
cljs.core.cst$sym$try = new cljs.core.Symbol(null,"try","try",-1273693247,null);
cljs.core.cst$sym$has_DASH_nil_QMARK_ = new cljs.core.Symbol(null,"has-nil?","has-nil?",825886722,null);
cljs.core.cst$sym$rear = new cljs.core.Symbol(null,"rear","rear",-900164830,null);
cljs.core.cst$sym$hierarchy = new cljs.core.Symbol(null,"hierarchy","hierarchy",587061186,null);
cljs.core.cst$sym$iter = new cljs.core.Symbol(null,"iter","iter",-1346195486,null);
cljs.core.cst$sym$namespace = new cljs.core.Symbol(null,"namespace","namespace",1263021155,null);
cljs.core.cst$sym$step = new cljs.core.Symbol(null,"step","step",-1365547645,null);
cljs.core.cst$sym$boolean = new cljs.core.Symbol(null,"boolean","boolean",-278886877,null);
cljs.core.cst$sym$update_DASH_count = new cljs.core.Symbol(null,"update-count","update-count",-411982269,null);
cljs.core.cst$sym$method_DASH_table = new cljs.core.Symbol(null,"method-table","method-table",-1878263165,null);
cljs.core.cst$sym$meta6766 = new cljs.core.Symbol(null,"meta6766","meta6766",-250516797,null);
cljs.core.cst$sym$chunk = new cljs.core.Symbol(null,"chunk","chunk",449371907,null);
cljs.core.cst$sym$i = new cljs.core.Symbol(null,"i","i",253690212,null);
cljs.core.cst$sym$rest = new cljs.core.Symbol(null,"rest","rest",398835108,null);
cljs.core.cst$kw$meta = new cljs.core.Keyword(null,"meta","meta",1499536964);
cljs.core.cst$kw$dup = new cljs.core.Keyword(null,"dup","dup",556298533);
cljs.core.cst$sym$comp = new cljs.core.Symbol(null,"comp","comp",-1462482139,null);
cljs.core.cst$sym$dispatch_DASH_fn = new cljs.core.Symbol(null,"dispatch-fn","dispatch-fn",-1401088155,null);
cljs.core.cst$kw$else = new cljs.core.Keyword(null,"else","else",-1508377146);
cljs.core.cst$kw$number = new cljs.core.Keyword(null,"number","number",1570378438);
cljs.core.cst$sym$left = new cljs.core.Symbol(null,"left","left",1241415590,null);
cljs.core.cst$kw$cljs$core_SLASH_none = new cljs.core.Keyword("cljs.core","none","cljs.core/none",926646439);
cljs.core.cst$sym$shift = new cljs.core.Symbol(null,"shift","shift",-1657295705,null);
cljs.core.cst$sym$iters = new cljs.core.Symbol(null,"iters","iters",719353031,null);
cljs.core.cst$sym$new_DASH_value = new cljs.core.Symbol(null,"new-value","new-value",-1567397401,null);
cljs.core.cst$sym$xform = new cljs.core.Symbol(null,"xform","xform",-85179481,null);
cljs.core.cst$kw$validator = new cljs.core.Keyword(null,"validator","validator",-1966190681);
cljs.core.cst$kw$default = new cljs.core.Keyword(null,"default","default",-1987822328);
cljs.core.cst$sym$prefer_DASH_table = new cljs.core.Symbol(null,"prefer-table","prefer-table",462168584,null);
cljs.core.cst$sym$loop_STAR_ = new cljs.core.Symbol(null,"loop*","loop*",615029416,null);
cljs.core.cst$sym$watches = new cljs.core.Symbol(null,"watches","watches",1367433992,null);
cljs.core.cst$kw$ns = new cljs.core.Keyword(null,"ns","ns",441598760);
cljs.core.cst$kw$name = new cljs.core.Keyword(null,"name","name",1843675177);
cljs.core.cst$sym$bitmap = new cljs.core.Symbol(null,"bitmap","bitmap",501334601,null);
cljs.core.cst$sym$meta7343 = new cljs.core.Symbol(null,"meta7343","meta7343",-1982384535,null);
cljs.core.cst$sym$_seq = new cljs.core.Symbol(null,"_seq","_seq",-449557847,null);
cljs.core.cst$sym$nil_DASH_val = new cljs.core.Symbol(null,"nil-val","nil-val",-513933559,null);
cljs.core.cst$sym$v = new cljs.core.Symbol(null,"v","v",1661996586,null);
cljs.core.cst$sym$riter = new cljs.core.Symbol(null,"riter","riter",-237834262,null);
cljs.core.cst$sym$__hash = new cljs.core.Symbol(null,"__hash","__hash",-1328796629,null);
cljs.core.cst$sym$meta = new cljs.core.Symbol(null,"meta","meta",-1154898805,null);
cljs.core.cst$sym$_meta = new cljs.core.Symbol(null,"_meta","_meta",-1716892533,null);
cljs.core.cst$sym$afn = new cljs.core.Symbol(null,"afn","afn",216963467,null);
cljs.core.cst$sym$tree = new cljs.core.Symbol(null,"tree","tree",1444219499,null);
cljs.core.cst$sym$fn = new cljs.core.Symbol(null,"fn","fn",465265323,null);
cljs.core.cst$sym$front = new cljs.core.Symbol(null,"front","front",117022539,null);
cljs.core.cst$sym$buf = new cljs.core.Symbol(null,"buf","buf",1426618187,null);
cljs.core.cst$sym$next_DASH_entry = new cljs.core.Symbol(null,"next-entry","next-entry",1091342476,null);
cljs.core.cst$kw$val = new cljs.core.Keyword(null,"val","val",128701612);
cljs.core.cst$sym$key = new cljs.core.Symbol(null,"key","key",124488940,null);
cljs.core.cst$sym$not_EQ_ = new cljs.core.Symbol(null,"not=","not=",1466536204,null);
cljs.core.cst$sym$_next = new cljs.core.Symbol(null,"_next","_next",101877036,null);
cljs.core.cst$sym$parent = new cljs.core.Symbol(null,"parent","parent",761652748,null);
cljs.core.cst$sym$root_DASH_iter = new cljs.core.Symbol(null,"root-iter","root-iter",1974672108,null);
cljs.core.cst$sym$validate = new cljs.core.Symbol(null,"validate","validate",1439230700,null);
cljs.core.cst$sym$do = new cljs.core.Symbol(null,"do","do",1686842252,null);
cljs.core.cst$sym$vec = new cljs.core.Symbol(null,"vec","vec",982683596,null);
cljs.core.cst$kw$fallback_DASH_impl = new cljs.core.Keyword(null,"fallback-impl","fallback-impl",-1501286995);
cljs.core.cst$kw$flush_DASH_on_DASH_newline = new cljs.core.Keyword(null,"flush-on-newline","flush-on-newline",-151457939);
cljs.core.cst$sym$default_DASH_dispatch_DASH_val = new cljs.core.Symbol(null,"default-dispatch-val","default-dispatch-val",-1231201266,null);
cljs.core.cst$kw$no_DASH_test = new cljs.core.Keyword(null,"no-test","no-test",-1679482642);
cljs.core.cst$sym$validator = new cljs.core.Symbol(null,"validator","validator",-325659154,null);
cljs.core.cst$sym$letfn_STAR_ = new cljs.core.Symbol(null,"letfn*","letfn*",-110097810,null);
cljs.core.cst$sym$if = new cljs.core.Symbol(null,"if","if",1181717262,null);
cljs.core.cst$sym$arr = new cljs.core.Symbol(null,"arr","arr",2115492975,null);
cljs.core.cst$sym$new = new cljs.core.Symbol(null,"new","new",-444906321,null);
cljs.core.cst$kw$descendants = new cljs.core.Keyword(null,"descendants","descendants",1824886031);
cljs.core.cst$sym$ns = new cljs.core.Symbol(null,"ns","ns",2082130287,null);
cljs.core.cst$kw$ancestors = new cljs.core.Keyword(null,"ancestors","ancestors",-776045424);
cljs.core.cst$sym$value = new cljs.core.Symbol(null,"value","value",1946509744,null);
cljs.core.cst$sym$name = new cljs.core.Symbol(null,"name","name",-810760592,null);
cljs.core.cst$sym$n = new cljs.core.Symbol(null,"n","n",-2092305744,null);
cljs.core.cst$kw$readably = new cljs.core.Keyword(null,"readably","readably",1129599760);
cljs.core.cst$kw$more_DASH_marker = new cljs.core.Keyword(null,"more-marker","more-marker",-14717935);
cljs.core.cst$sym$fields = new cljs.core.Symbol(null,"fields","fields",-291534703,null);
cljs.core.cst$sym$method_DASH_cache = new cljs.core.Symbol(null,"method-cache","method-cache",1230193905,null);
cljs.core.cst$sym$nexts = new cljs.core.Symbol(null,"nexts","nexts",-1683579407,null);
cljs.core.cst$sym$edit = new cljs.core.Symbol(null,"edit","edit",-1302639,null);
cljs.core.cst$sym$editable_QMARK_ = new cljs.core.Symbol(null,"editable?","editable?",-164945806,null);
cljs.core.cst$sym$base_DASH_count = new cljs.core.Symbol(null,"base-count","base-count",-1180647182,null);
cljs.core.cst$sym$collision_DASH_hash = new cljs.core.Symbol(null,"collision-hash","collision-hash",-35831342,null);
cljs.core.cst$sym$deftype_STAR_ = new cljs.core.Symbol(null,"deftype*","deftype*",962659890,null);
cljs.core.cst$sym$let_STAR_ = new cljs.core.Symbol(null,"let*","let*",1920721458,null);
cljs.core.cst$sym$start = new cljs.core.Symbol(null,"start","start",1285322546,null);
cljs.core.cst$sym$coll = new cljs.core.Symbol(null,"coll","coll",-1006698606,null);
cljs.core.cst$sym$not_DASH_native = new cljs.core.Symbol(null,"not-native","not-native",-236392494,null);
cljs.core.cst$sym$js_STAR_ = new cljs.core.Symbol(null,"js*","js*",-1134233646,null);
cljs.core.cst$sym$strobj = new cljs.core.Symbol(null,"strobj","strobj",1088091283,null);
cljs.core.cst$sym$_rest = new cljs.core.Symbol(null,"_rest","_rest",-2100466189,null);
cljs.core.cst$sym$fn_STAR_ = new cljs.core.Symbol(null,"fn*","fn*",-752876845,null);
cljs.core.cst$sym$val = new cljs.core.Symbol(null,"val","val",1769233139,null);
cljs.core.cst$sym$ascending_QMARK_ = new cljs.core.Symbol(null,"ascending?","ascending?",-1938452653,null);
cljs.core.cst$sym$meta5749 = new cljs.core.Symbol(null,"meta5749","meta5749",1568089939,null);
cljs.core.cst$sym$recur = new cljs.core.Symbol(null,"recur","recur",1202958259,null);
cljs.core.cst$sym$ci = new cljs.core.Symbol(null,"ci","ci",2049808339,null);
cljs.core.cst$kw$print_DASH_length = new cljs.core.Keyword(null,"print-length","print-length",1931866356);
cljs.core.cst$sym$state = new cljs.core.Symbol(null,"state","state",-348086572,null);
cljs.core.cst$kw$ok = new cljs.core.Keyword(null,"ok","ok",967785236);
cljs.core.cst$sym$cached_DASH_hierarchy = new cljs.core.Symbol(null,"cached-hierarchy","cached-hierarchy",-1085460203,null);
cljs.core.cst$sym$s = new cljs.core.Symbol(null,"s","s",-948495851,null);
cljs.core.cst$kw$parents = new cljs.core.Keyword(null,"parents","parents",-2027538891);
cljs.core.cst$sym$cnt = new cljs.core.Symbol(null,"cnt","cnt",1924510325,null);
cljs.core.cst$sym$node = new cljs.core.Symbol(null,"node","node",-2073234571,null);
cljs.core.cst$sym$sym = new cljs.core.Symbol(null,"sym","sym",195671222,null);
cljs.core.cst$sym$sb = new cljs.core.Symbol(null,"sb","sb",-1249746442,null);
cljs.core.cst$sym$meta6763 = new cljs.core.Symbol(null,"meta6763","meta6763",1110198296,null);
cljs.core.cst$sym$_hash = new cljs.core.Symbol(null,"_hash","_hash",-2130838312,null);
cljs.core.cst$sym$meta5746 = new cljs.core.Symbol(null,"meta5746","meta5746",216006520,null);
cljs.core.cst$sym$keys = new cljs.core.Symbol(null,"keys","keys",-1586012071,null);
cljs.core.cst$sym$set_BANG_ = new cljs.core.Symbol(null,"set!","set!",250714521,null);
cljs.core.cst$kw$tag = new cljs.core.Keyword(null,"tag","tag",-1290361223);
cljs.core.cst$sym$tree_DASH_map = new cljs.core.Symbol(null,"tree-map","tree-map",1373073049,null);
cljs.core.cst$sym$_DOT_ = new cljs.core.Symbol(null,".",".",1975675962,null);
cljs.core.cst$kw$mutable = new cljs.core.Keyword(null,"mutable","mutable",875778266);
cljs.core.cst$sym$quote = new cljs.core.Symbol(null,"quote","quote",1377916282,null);
cljs.core.cst$sym$root = new cljs.core.Symbol(null,"root","root",1191874074,null);
cljs.core.cst$sym$str = new cljs.core.Symbol(null,"str","str",-1564826950,null);
cljs.core.cst$sym$nodes = new cljs.core.Symbol(null,"nodes","nodes",-459054278,null);
cljs.core.cst$sym$seen = new cljs.core.Symbol(null,"seen","seen",1121531738,null);
cljs.core.cst$sym$hash_DASH_map = new cljs.core.Symbol(null,"hash-map","hash-map",-439030950,null);
cljs.core.cst$kw$arglists = new cljs.core.Keyword(null,"arglists","arglists",1661989754);
cljs.core.cst$sym$nil_DASH_iter = new cljs.core.Symbol(null,"nil-iter","nil-iter",1101030523,null);
cljs.core.cst$kw$alt_DASH_impl = new cljs.core.Keyword(null,"alt-impl","alt-impl",670969595);
cljs.core.cst$sym$ext_DASH_map_DASH_iter = new cljs.core.Symbol(null,"ext-map-iter","ext-map-iter",-1215982757,null);
cljs.core.cst$sym$tail = new cljs.core.Symbol(null,"tail","tail",494507963,null);
cljs.core.cst$sym$record = new cljs.core.Symbol(null,"record","record",861424668,null);
cljs.core.cst$sym$mseq = new cljs.core.Symbol(null,"mseq","mseq",1602647196,null);
cljs.core.cst$sym$count = new cljs.core.Symbol(null,"count","count",-514511684,null);
cljs.core.cst$kw$keywordize_DASH_keys = new cljs.core.Keyword(null,"keywordize-keys","keywordize-keys",1310784252);
cljs.core.cst$sym$off = new cljs.core.Symbol(null,"off","off",-2047994980,null);
cljs.core.cst$sym$stack = new cljs.core.Symbol(null,"stack","stack",847125597,null);
cljs.core.cst$sym$transient_DASH_map = new cljs.core.Symbol(null,"transient-map","transient-map",351764893,null);
cljs.core.cst$sym$meta7340 = new cljs.core.Symbol(null,"meta7340","meta7340",-1789812195,null);
cljs.core.cst$sym$len = new cljs.core.Symbol(null,"len","len",-1230778691,null);
cljs.core.cst$sym$right = new cljs.core.Symbol(null,"right","right",1187949694,null);
cljs.core.cst$sym$throw = new cljs.core.Symbol(null,"throw","throw",595905694,null);
cljs.core.cst$sym$fseq = new cljs.core.Symbol(null,"fseq","fseq",-1466412450,null);
cljs.core.cst$sym$number_QMARK_ = new cljs.core.Symbol(null,"number?","number?",-1747282210,null);
cljs.core.cst$sym$symbol_QMARK_ = new cljs.core.Symbol(null,"symbol?","symbol?",1820680511,null);
cljs.core.cst$kw$cljs$core_SLASH_not_DASH_found = new cljs.core.Keyword("cljs.core","not-found","cljs.core/not-found",-1572889185);
cljs.core.cst$sym$more = new cljs.core.Symbol(null,"more","more",-418290273,null);
cljs.core.cst$sym$def = new cljs.core.Symbol(null,"def","def",597100991,null);
cljs.core.cst$kw$data = new cljs.core.Keyword(null,"data","data",-232669377);
cljs.core.cst$sym$f = new cljs.core.Symbol(null,"f","f",43394975,null);
cljs.core.cst$sym$next_DASH_iter = new cljs.core.Symbol(null,"next-iter","next-iter",1526626239,null);


out/js/imei/core.js
// Compiled by ClojureScript 1.7.145 {:static-fns true}
goog.provide('imei.core');
goog.require('cljs.core');
goog.require('goog.dom');
imei.core.span_n_sims = document.getElementById("n-sims");
imei.core.span_network_name = document.getElementById("network-name");
imei.core.span_network_signal = document.getElementById("network-signal");
imei.core.span_network_mcc = document.getElementById("network-mcc");
imei.core.span_network_mnc = document.getElementById("network-mnc");
imei.core.span_network_supported = document.getElementById("network-supported");
imei.core.span_iccid = document.getElementById("iccid");
imei.core.a_sync = document.getElementById("sync");
imei.core.span_network_type = document.getElementById("network-type");
imei.core.span_network_state = document.getElementById("network-state");
imei.core.span_cellid = document.getElementById("cellid");
imei.core.span_location_area = document.getElementById("location-area");
imei.core.span_wifi_mac = document.getElementById("wifi-mac");
imei.core.span_wifi_ip = document.getElementById("wifi-ip");
imei.core.btn_imei = document.getElementById("btn-imei");
imei.core.btn_msisdn = document.getElementById("btn-msisdn");
imei.core.btn_forward = document.getElementById("btn-forward");
imei.core.start = (function imei$core$start(){
imei.core.supportedNetworkTypes_str = window.navigator.mozMobileConnections[0].supportedNetworkTypes.join(' - ');

var G__5775_5799 = imei.core.span_n_sims;
var G__5776_5800 = window.navigator.mozMobileConnections.length;
goog.dom.setTextContent(G__5775_5799,G__5776_5800);

var G__5777_5801 = imei.core.span_network_name;
var G__5778_5802 = window.navigator.mozMobileConnections[0].voice.network.longName;
goog.dom.setTextContent(G__5777_5801,G__5778_5802);

var G__5779_5803 = imei.core.span_network_mcc;
var G__5780_5804 = window.navigator.mozMobileConnections[0].voice.network.mcc;
goog.dom.setTextContent(G__5779_5803,G__5780_5804);

var G__5781_5805 = imei.core.span_network_mnc;
var G__5782_5806 = window.navigator.mozMobileConnections[0].voice.network.mnc;
goog.dom.setTextContent(G__5781_5805,G__5782_5806);

var G__5783_5807 = imei.core.span_network_signal;
var G__5784_5808 = window.navigator.mozMobileConnections[0].voice.relSignalStrength;
goog.dom.setTextContent(G__5783_5807,G__5784_5808);

goog.dom.setTextContent(imei.core.span_network_supported,imei.core.supportedNetworkTypes_str);

var G__5785_5809 = imei.core.span_iccid;
var G__5786_5810 = window.navigator.mozIccManager.iccIds[0];
goog.dom.setTextContent(G__5785_5809,G__5786_5810);

var G__5787_5811 = imei.core.span_network_type;
var G__5788_5812 = window.navigator.mozMobileConnections[0].data.type;
goog.dom.setTextContent(G__5787_5811,G__5788_5812);

var G__5789_5813 = imei.core.span_network_state;
var G__5790_5814 = window.navigator.mozMobileConnections[0].data.state;
goog.dom.setTextContent(G__5789_5813,G__5790_5814);

var G__5791_5815 = imei.core.span_cellid;
var G__5792_5816 = window.navigator.mozMobileConnections[0].data.cell.gsmCellId;
goog.dom.setTextContent(G__5791_5815,G__5792_5816);

var G__5793_5817 = imei.core.span_location_area;
var G__5794_5818 = window.navigator.mozMobileConnections[0].data.cell.gsmLocationAreaCode;
goog.dom.setTextContent(G__5793_5817,G__5794_5818);

var G__5795_5819 = imei.core.span_wifi_mac;
var G__5796_5820 = window.navigator.mozWifiManager.macAddress;
goog.dom.setTextContent(G__5795_5819,G__5796_5820);

var G__5797 = imei.core.span_wifi_ip;
var G__5798 = window.navigator.mozWifiManager.connectionInformation.ipAddress;
return goog.dom.setTextContent(G__5797,G__5798);
});
imei.core.handle_imei = (function imei$core$handle_imei(){
console.log("imei");

return (new MozActivity(cljs.core.clj__GT_js(new cljs.core.PersistentArrayMap(null, 2, [new cljs.core.Keyword(null,"name","name",1843675177),"dial",new cljs.core.Keyword(null,"data","data",-232669377),new cljs.core.PersistentArrayMap(null, 1, [new cljs.core.Keyword(null,"number","number",1570378438),"*#06#"], null)], null))));
});
imei.core.handle_forward = (function imei$core$handle_forward(){
return (new MozActivity(cljs.core.clj__GT_js(new cljs.core.PersistentArrayMap(null, 2, [new cljs.core.Keyword(null,"name","name",1843675177),"dial",new cljs.core.Keyword(null,"data","data",-232669377),new cljs.core.PersistentArrayMap(null, 1, [new cljs.core.Keyword(null,"number","number",1570378438),"*#21#"], null)], null))));
});
imei.core.handle_msisdn = (function imei$core$handle_msisdn(){
return navigator.getMobileIdAssertion({forceSelection: true});
});
imei.core.btn_imei.addEventListener("click",imei.core.handle_imei);
imei.core.btn_forward.addEventListener("click",imei.core.handle_forward);
imei.core.btn_msisdn.addEventListener("click",imei.core.handle_msisdn);
imei.core.a_sync.addEventListener("click",imei.core.start);
window.onload = imei.core.start;
